caluma/caluma_form/jexl.py

```diff
@@ -109,18 +120,19 @@ def use_field_context(self, field: Field):
         self._structure = old_structure
 
     def _get_referenced_fields(self, field: Field, expr: str):
-        deps = list(self.extract_referenced_questions(expr))
-        referenced_fields = [self._structure.get_field(slug) for slug in deps]
+        deps = list(self.extract_referenced_questions_with_arguments(expr))
+        referenced_fields = [self._structure.get_field(slug) for slug, _ in deps]
 
         referenced_slugs = [ref.question.slug for ref in referenced_fields if ref]
 
-        for slug in deps:
-            if slug not in referenced_slugs:
+        for slug, args in deps:
+            required = not len(args) or (args[0] and args[0].value)
```

Comment:
I think an explanatory comment would be helpful: Why is having args impacting the requiredness? In the context of the PR it's clear, but future code readers might get confused

Reply:
> Why is having args impacting the requiredness?

That statement is not entirely correct. It's not having any args that impacting the requiredness, it's the first argument specifically that needs to be `False` to impact the requiredness. This behaviour is clearly documented in the user docs. I really think this line right here is self-explanatory and doesn't need a comment.